astroquery/mast/core.py

```diff
@@ -0,0 +1,398 @@
+            E.g. when using a slow internet connection.
+        page : int or None (optional)
+            Default None. 
+            Can be used to override the default behavior of all results being returned to obtain 
+            a sepcific page of results.
+        verbose : bool (optional)
+            Default False. Setting to True provides more extensive output.
+        **kwargs: 
+            See MashupRequest properties `here <https://masttest.stsci.edu/api/v0/class_mashup_1_1_mashup_request.html>`_ for additional keyword arguments.
+        
+        
+        Returns
+        -------
+            response: list(`requests.Response`)
+        """
+        
+        # setting up pagination
+        if not pagesize:
+            pagesize=self.PAGESIZE
+        if not page:
+            page=1
+            retrieveAll = True
+        else:
+            retrieveAll = False
+        
+        headers = {"User-Agent":self._session.headers["User-Agent"],
+                   "Content-type": "application/x-www-form-urlencoded",
+                   "Accept": "text/plain"}
+        
+    
+        mashupRequest = {'service':service,
+                         'params':params,
+                         'format':'json',
+                         'pagesize':pagesize, 
+                         'page':page}
+    
+        reqString = _prepare_service_request_string(mashupRequest)
+        response = self._request("POST",self._SERVER+"/api/v0/invoke",data=reqString,headers=headers,
+                                 retrieve_all=retrieveAll,verbose=verbose)
+        
+        return response
+        
+        
+    def _resolve_object(self,objectname,verbose=False):
+        """
+        Resolves an object name to a position on the sky.
+        
+        Parameters
+        ----------
+        objectname : str
+            Name of astronimical object to resolve.
+        verbose : bool (optional)
+            Default False. Setting to True provides more extensive output.    
+        """
+        
+        service = 'Mast.Name.Lookup'
+        params ={'input':objectname,
+                 'format':'json'}
+        
+        response = self.service_request_async(service,params)
+        
+        result = response[0].json() 
+        
+        if len(result['resolvedCoordinate']) == 0:
+            raise ResolverError("Could not resolve %s to a sky position." % objectname)
+            
+        
+        ra = result['resolvedCoordinate'][0]['ra']
+        dec = result['resolvedCoordinate'][0]['decl']
+        coordinates = coord.SkyCoord(ra, dec, unit="deg")
+        
+        return coordinates
+
+
+
+@async_to_sync
+class ObservationsClass(MastClass):
+    """
+    MAST Observations query class.
+
+    Class for querying MAST observational data.
+    """
+    
+        
+    @class_or_instance
+    def query_region_async(self, coordinates, radius="0.2 deg", pagesize=None, page=None, verbose=False):
```

Comment:
We prefer to use astropy objects over string units here, so `radius=0.2*u.deg` instead of `"0.2 deg"`